l1-contracts/test/benchmark/happy.t.sol

```diff
@@ -0,0 +1,413 @@
+import {NaiveMerkle} from "../merkle/Naive.sol";
+import {MerkleTestUtil} from "../merkle/TestUtil.sol";
+import {TestERC20} from "@aztec/mock/TestERC20.sol";
+import {TestConstants} from "../harnesses/TestConstants.sol";
+import {RewardDistributor} from "@aztec/governance/RewardDistributor.sol";
+import {IERC20Errors} from "@oz/interfaces/draft-IERC6093.sol";
+import {IFeeJuicePortal} from "@aztec/core/interfaces/IFeeJuicePortal.sol";
+import {IRewardDistributor, IRegistry} from "@aztec/governance/interfaces/IRewardDistributor.sol";
+import {ProposeArgs, OracleInput, ProposeLib} from "@aztec/core/libraries/rollup/ProposeLib.sol";
+import {IERC20} from "@oz/token/ERC20/IERC20.sol";
+import {
+  FeeLib,
+  FeeAssetPerEthE9,
+  EthValue,
+  FeeHeader,
+  L1FeeData,
+  ManaBaseFeeComponents
+} from "@aztec/core/libraries/rollup/FeeLib.sol";
+import {CheatDepositArgs} from "@aztec/core/interfaces/IRollup.sol";
+import {
+  FeeModelTestPoints,
+  TestPoint,
+  FeeHeaderModel,
+  ManaBaseFeeComponentsModel
+} from "test/fees/FeeModelTestPoints.t.sol";
+import {MessageHashUtils} from "@oz/utils/cryptography/MessageHashUtils.sol";
+import {
+  Timestamp, Slot, Epoch, SlotLib, EpochLib, TimeLib
+} from "@aztec/core/libraries/TimeLib.sol";
+import {Forwarder} from "@aztec/periphery/Forwarder.sol";
+
+// solhint-disable comprehensive-interface
+
+uint256 constant MANA_TARGET = 100000000;
+uint256 constant VALIDATOR_COUNT = 100;
+
+contract FakeCanonical is IRewardDistributor {
+  uint256 public constant BLOCK_REWARD = 50e18;
+  IERC20 public immutable UNDERLYING;
+
+  address public canonicalRollup;
+
+  constructor(IERC20 _asset) {
+    UNDERLYING = _asset;
+  }
+
+  function setCanonicalRollup(address _rollup) external {
+    canonicalRollup = _rollup;
+  }
+
+  function claim(address _recipient) external returns (uint256) {
+    TestERC20(address(UNDERLYING)).mint(_recipient, BLOCK_REWARD);
+    return BLOCK_REWARD;
+  }
+
+  function claimBlockRewards(address _recipient, uint256 _blocks) external returns (uint256) {
+    TestERC20(address(UNDERLYING)).mint(_recipient, _blocks * BLOCK_REWARD);
+    return _blocks * BLOCK_REWARD;
+  }
+
+  function distributeFees(address _recipient, uint256 _amount) external {
+    TestERC20(address(UNDERLYING)).mint(_recipient, _amount);
+  }
+
+  function updateRegistry(IRegistry _registry) external {}
+}
+
+contract BenchmarkRollupTest is FeeModelTestPoints, DecoderBase {
+  using MessageHashUtils for bytes32;
+  using stdStorage for StdStorage;
+  using TimeLib for Slot;
+
+  using SlotLib for Slot;
+  using EpochLib for Epoch;
+  using FeeLib for uint256;
+  using FeeLib for ManaBaseFeeComponents;
+  // We need to build a block that we can submit. We will be using some values from
+  // the empty blocks, but otherwise populate using the fee model test points.
+
+  struct Block {
+    ProposeArgs proposeArgs;
+    bytes blobInputs;
+    Signature[] signatures;
+  }
+
+  DecoderBase.Full full = load("empty_block_1");
```

Comment:
We would benefit from a variant with a non empty block

Reply:
I'm not sure it would make a huge diff in these cases since the blobs are not actually published, and the mana is actually loaded in from the testdata from the fees earlier. So it is not "really" an empty one. I guess the `txHashes` is the only that would differ much 🤔